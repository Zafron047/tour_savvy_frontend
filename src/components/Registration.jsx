import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { registrationUser } from '../redux/authentication/userSlice';
import '../stylesheets/registration.css';

const Registration = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const loading = useSelector((state) => state.current_user.loading);
  const error = useSelector((state) => state.current_user.error);
  const [formData, setFormData] = useState({
    username: '',
    email: '',
    password: '',
  });
  const handleChange = (e) => {
    setFormData({
      ...formData,
      [e.target.name]: e.target.value,
      [e.target.email]: e.target.value,
      [e.target.password]: e.target.value,
    });
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(registrationUser({
      user:
      {
        username: formData.username,
        email: formData.email,
        password: formData.password,
      },
    })).then((response) => {
      if (response.payload && response.payload.username) {
        navigate('/');
      } else {
        alert('You should put a user name');
      }
    }).catch((error) => {
      alert('Error during registration', error.message);
    });
  };
  return (
    <div className="regi-bg">
      <div className="registration-container">
        <h2 className="registration-title">Registration</h2>
        {error && <p className="error">{error}</p>}
        <form onSubmit={handleSubmit}>
          <div className="form-group">
            <input
              type="text"
              id="username"
              name="username"
              placeholder="Username"
              value={formData.username}
              onChange={handleChange}
              className="input-field"
            />
          </div>
          <div className="form-group">
            <input
              type="text"
              id="email"
              name="email"
              placeholder="Email"
              value={formData.email}
              onChange={handleChange}
              className="input-field"
            />
          </div>
          <div className="form-group">
            <input
              type="text"
              id="password"
              name="password"
              placeholder="Password"
              value={formData.password}
              onChange={handleChange}
              className="input-field"
            />
          </div>
          <div className="form-group">
            <button type="submit" disabled={loading} className="submit-button">
              {loading ? 'Registering...' : 'Register'}
            </button>
          </div>
        </form>
      </div>
    </div>
  );
};

export default Registration;
