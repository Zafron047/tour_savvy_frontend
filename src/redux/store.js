import { configureStore } from '@reduxjs/toolkit';
import packagesSlice from './packages/packagesSlice';
import reservationsReducer from './reservations/reservationSlice';
import userSlice from './authentication/userSlice';

const loadStateFromLocalStorage = () => {
  try {
    const serializedState = localStorage.getItem('reduxState');
    if (serializedState === null) {
      return undefined;
    }
    return JSON.parse(serializedState);
  } catch (err) {
    return undefined;
  }
};

const loadUserFromLocalStorage = () => {
  try {
    const serializedUser = localStorage.getItem('user');
    if (serializedUser === null) {
      return undefined;
    }
    return JSON.parse(serializedUser);
  } catch (err) {
    return undefined;
  }
};

const store = configureStore({
  reducer: {
    packages: packagesSlice,
    reservations: reservationsReducer,
    current_user: userSlice,
  },
  preloadedState: {
    current_user: loadUserFromLocalStorage(),
    ...loadStateFromLocalStorage(),
  },
  reservations: reservationsReducer,
});

export default store;
