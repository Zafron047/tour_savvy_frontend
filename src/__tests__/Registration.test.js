import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Registration from '../components/Registration';

test('Should render Registration component correctly', () => {
  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Registration />
      </BrowserRouter>
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
