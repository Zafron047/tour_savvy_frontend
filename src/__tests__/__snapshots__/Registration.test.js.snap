// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Registration component correctly 1`] = `
<div>
  <div>
    <h2>
      Registration
    </h2>
    <form>
      <div>
        <input
          id="username"
          name="username"
          placeholder="Username"
          type="text"
          value=""
        />
      </div>
      <div>
        <input
          id="email"
          name="email"
          placeholder="Email"
          type="text"
          value=""
        />
      </div>
      <div>
        <input
          id="password"
          name="password"
          placeholder="Password"
          type="text"
          value=""
        />
      </div>
      <div>
        <button
          type="submit"
        >
          Register
        </button>
      </div>
    </form>
  </div>
</div>
`;
